### PR TITLE
Added googlemock and googletest

### PR DESCRIPTION
**Pull request by Norfolc**
Added googlemock.
Added file_sdk test that uses googletest.
Commented out rms unit tests in scons since they cause build errors.

### CODE REVIEW

File: googlemock/SConscript [hunk 1]
```diff
@@ -0,0 +1,12 @@
+Import('env')
+
+src_files = Split("""
+	src/gmock-all.cc
+""")
+env.Append(CPPPATH='#googlemock')
+env.Append(CPPPATH='#googletest')
+env.Append(CPPPATH='#googlemock/include')
+env.Append(CPPPATH='#googletest/include')
```
Comment (assapir) [inline googlemock/SConscript]: use split #Resolved

File: sdk/file_sdk/UnitTests/gtest/SConscript [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+src_files = ['unittest_main.cpp']
+
+'''
+src_files += Glob('#src/upe/api_impl/test/*.cpp')
+src_files += Glob('#src/upe/base/test/*.cpp')
+src_files += Glob('#src/upe/common/test/*.cpp')
+src_files += Glob('#src/upe/core/test/*.cpp')
+src_files += Glob('#src/upe/core/parser/test/*.cpp')
+src_files += Glob('#src/upe/store/test/*.cpp')
+src_files += Glob('#src/shared/telemetry/test/*.cpp')
```
Comment (assapir) [inline sdk/file_sdk/UnitTests/gtest/SConscript]: why include things that are not there and will probably will be in other path? #Resolved

File: sdk/file_sdk/UnitTests/SConscript [hunk 3]
```diff
@@ -1,3 +1,8 @@
-Import('file_env')
+Import('file_env bins')
 
-common_ut_lib = file_env.SConscript('common_ut/SConscript', duplicate=0)
\ No newline at end of file
+variant_dir = '#' +  bins + '/sdk/file_sdk/unittests/'
+
+#common_ut_lib = file_env.SConscript('common_ut/SConscript', variant_dir=variant_dir + 'common', duplicate=0)
```
Comment (assapir) [inline sdk/file_sdk/UnitTests/SConscript]: >on_ut_lib = file_env.SConscript('common_ut/SConscript', variant_dir=variant_dir + 'common', duplicate=0) [](start = 5, length = 104)

just remove until fixed #Resolved

File: sdk/file_sdk/UnitTests/gtest/SConscript (hunk 2 above)
Comment (Norfolc) [reply sdk/file_sdk/UnitTests/gtest/SConscript]: these are commented out... will remove it

---
In reply to: [139422602](https://github.com/AzureAD/rms-sdk-for-cpp/pull/161#discussion_r139422602) [](ancestors = 139422602)

**Review by assapir — APPROVED** [review]
:shipit:

**Review by shenhavba — APPROVED** [review]
:shipit:

**Pull request merged**